src/Structure/Content/Menu.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Abstract class for a custom menu.
+ *
+ * @author  Chrispian Burks <chrispian.burks@webdevstudios.com>
+ * @package WebDevStudios\OopsWP\Structure\Content
+ */
+
+namespace WebDevStudios\OopsWP\Structure\Content;
+
+/**
+ * Class Menu
+ *
+ * @package WebDevStudios\OopsWP\Structure\Content
+ * @since   2019-05-01
+ */
+abstract class Menu implements MenuInterface {
+	/**
+	 * Menu Location.
+	 *
+	 * @var string
+	 * @since 2019-05-01
+	 */
+	protected $location;
+
+	/**
+	 * Menu Description.
+	 *
+	 * @var string
+	 * @since 2019-05-01
+	 */
+	protected $description;
+
+	/**
+	 * Callback to register the menu with WordPress.
+	 *
+	 * @author Chrispian H. Burks <chrispian.burks@webdevstudios.com>
+	 * @throws \Exception If validation fails throw an exception.
+	 * @since 2019-05-01
+	 */
+	public function register() {
+		$this->validate_location();
+		$this->validate_description();
+
+		register_nav_menu( $this->location, $this->description );
+	}
+
+	/**
+	 * Validate the provided menu location.
+	 *
+	 * @author Chrispian Burks <chrispian.burks@webdevstudios.com>
+	 * @throws \Exception If the menu location is not valid.
+	 * @since  2019-05-01
+	 */
+	private function validate_location() {
+		if ( empty( $this->location ) ) {
+			throw new \Exception( 'You must define the $location property for your menu in ' . __CLASS__ );
+		}
+
+		if ( ! is_string( $this->location ) ) {
+			throw new \Exception( 'The $location property defined in your menu class ' . __CLASS__ . ' must be a string' );
+		}
+	}
+
+	/**
+	 * Validate the provided menu description.
+	 *
+	 * @author Chrispian Burks <chrispian.burks@webdevstudios.com>
+	 * @throws \Exception If the menu description is not valid.
+	 * @since  2019-05-01
+	 */
+	private function validate_description() {
+		if ( empty( $this->description ) ) {
+			throw new \Exception( 'You must define the $description property for your menu in ' . __CLASS__ );
+		}
+
+		if ( ! is_string( $this->description ) ) {
```

Comment:
Same situation here. 

In fact, you could probably consolidate these two checks into the `check_requirements` method. It might look like this:

```
foreach ( [ 'location', 'description' ] as $property ) {
    if ( ! is_string( $this->{$property} ) {
        throw new Exception( '${$property} must be of type string in ' . __CLASS__ );
    }
}
```

That would let you cut both of the `validate` methods to simplify this a bit.

Reply:
Also, in #25 I was throwing an `UnexpectedValueException`, which might be more accurate here. Adding a `use` statement below the namespace will let you call it without the prefixed `\`.

Reply:
I love how much simpler this is. I've also changed to UnexpectedValueException, that seems more accurate here. Let me know if that's not correct (pushing updates soon).